cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -119,32 +119,53 @@ func (o *ScaleUpOrchestrator) ScaleUp(
 		}
 	}
 
+	var nodeGroupPoolList [][]cloudprovider.NodeGroup
+	// Add cloudprovider nodegroup pool into the list.
+	nodeGroupPoolList = append(nodeGroupPoolList, nodeGroups)
+
+	if o.processors != nil && o.processors.NodeGroupPoolListProcessor != nil {
+		var err error
+		nodeGroupPoolList, nodeInfos, err = o.processors.NodeGroupPoolListProcessor.Process(o.autoscalingContext, nodeGroupPoolList, nodeInfos, unschedulablePods)
+		if err != nil {
+			return scaleUpError(&status.ScaleUpStatus{}, errors.ToAutoscalerError(errors.InternalError, err))
+		}
+	}
+
 	resourcesLeft, aErr := o.resourceManager.ResourcesLeft(o.autoscalingContext, nodeInfos, nodes)
 	if aErr != nil {
 		return scaleUpError(&status.ScaleUpStatus{}, aErr.AddPrefix("could not compute total resources: "))
 	}
 
 	now := time.Now()
 
-	// Filter out invalid node groups
-	validNodeGroups, skippedNodeGroups := o.filterValidScaleUpNodeGroups(nodeGroups, nodeInfos, resourcesLeft, now)
-
-	// Calculate expansion options
-	schedulablePods := map[string][]*apiv1.Pod{}
+	var validNodeGroups []cloudprovider.NodeGroup
+	var skippedNodeGroups map[string]status.Reasons
 	var options []expander.Option
+	schedulablePods := map[string][]*apiv1.Pod{}
 
-	for _, nodeGroup := range validNodeGroups {
-		schedulablePods[nodeGroup.Id()] = o.SchedulablePods(podEquivalenceGroups, nodeGroup, nodeInfos[nodeGroup.Id()])
-	}
+	for _, nodeGroups = range nodeGroupPoolList {
+		// Filter out invalid node groups
+		validNodeGroups, skippedNodeGroups = o.filterValidScaleUpNodeGroups(nodeGroups, nodeInfos, resourcesLeft, now)
```

Comment:
We cannot override `skippedNodeGroups` every loop, this is used for Events about the scheduling. We should join the result with the outer map instead.

Reply:
Done!